kedro-datasets/kedro_datasets/pandas/deltatable_dataset.py

```diff
@@ -0,0 +1,258 @@
+
+    .. code-block:: yaml
+
+        boats_filesystem:
+          type: pandas.DeltaTableDataSet
+          filepath: data/01_raw/boats
+          credentials: dev_creds
+          load_args:
+            version: 7
+          save_args:
+            mode: overwrite
+
+        boats_databricks_unity_catalog:
+          type: pandas.DeltaTableDataSet
+          credentials: dev_creds
+          catalog_type: UNITY
+          database: simple_database
+          table: simple_table
+          save_args:
+            mode: overwrite
+
+        trucks_aws_glue_catalog:
+          type: pandas.DeltaTableDataSet
+          credentials: dev_creds
+          catalog_type: AWS
+          catalog_name: main
+          database: db_schema
+          table: db_table
+          save_args:
+            mode: overwrite
+
+    Example usage for the `Python API`_:
+    ::
+
+        >>> from kedro_datasets.pandas import DeltaTableDataSet
+        >>> import pandas as pd
+        >>>
+        >>> data = pd.DataFrame({'col1': [1, 2], 'col2': [4, 5], 'col3': [5, 6]})
+        >>> data_set = DeltaTableDataSet(filepath="test")
+        >>>
+        >>> data_set.save(data)
+        >>> reloaded = data_set.load()
+        >>> assert data.equals(reloaded)
+        >>>
+        >>> new_data = pd.DataFrame({'col1': [7, 8], 'col2': [9, 10], 'col3': [11, 12]})
+        >>> data_set.save(new_data)
+        >>> data_set.get_loaded_version()
+
+    """
+
+    DEFAULT_WRITE_MODE = "overwrite"
+    ACCEPTED_WRITE_MODES = ("overwrite", "append")
+
+    DEFAULT_LOAD_ARGS: Dict[str, Any] = {}
+    DEFAULT_SAVE_ARGS: Dict[str, Any] = {"mode": DEFAULT_WRITE_MODE}
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        filepath: Optional[str] = None,
+        catalog_type: Optional[DataCatalog] = None,
+        catalog_name: Optional[str] = None,
+        database: Optional[str] = None,
+        table: Optional[str] = None,
+        load_args: Optional[Dict[str, Any]] = None,
+        save_args: Optional[Dict[str, Any]] = None,
+        credentials: Optional[Dict[str, Any]] = None,
+        fs_args: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """Creates a new instance of ``DeltaTableDataSet``
+
+        Args:
+            filepath (str): Filepath to a delta lake file with the following accepted protocol:
+                ``S3``:
+                    `s3://<bucket>/<path>`
+                    `s3a://<bucket>/<path>`
+                ``Azure``:
+                    `az://<container>/<path>`
+                    `adl://<container>/<path>`
+                    `abfs://<container>/<path>`
+                ``GCS``:
+                    `gs://<bucket>/<path>`
+                If any of the prefix above is not provided, `file` protocol (local filesystem)
+                will be used.
+            catalog_type (DataCatalog, optional): `AWS` or `UNITY` if filepath is not provided.
+                Defaults to None.
```

Comment:
What does this catalog_type do?

Reply:
`catalog_type` loads tables from either `glue` (aws) or `unity` (databricks) if specified, reference here: https://delta-io.github.io/delta-rs/python/usage.html#loading-a-delta-table